tests/spicy/types/unit/basic-incremental.spicy

```diff
@@ -1,7 +1,7 @@
 # @TEST-EXEC:  ${SCRIPTS}/printf '\x00\x00\x27\x10\x2a' | HILTI_DEBUG=spicy:spicy-driver spicy-driver -d -i 1 %INPUT >output
 # @TEST-EXEC:  btest-diff output
 # @TEST-EXEC:  test "$BUILD" != "debug" || (cat .stderr | grep "\[spicy\] " >debug.log && btest-diff debug.log)
-# @TEST-EXEC:  test "$BUILD" != "debug" || (cat .stderr | grep -q "\[spicy-driver\] input.*chunks-cur=0 offset-head=4 offset-tail=4")
+# @TEST-EXEC:  test "$BUILD" != "debug" || (cat .stderr | grep -q "\[spicy-driver\] input.*chunks-cur=4 offset-head=0 offset-tail=4")
```

Comment:
Does this mean we are breaking incremental parsing with the original change @rsmmr?

Reply:
FTR, the output we are seeing with the changes in this PR is

```
[spicy-driver] input : size-current=0 size-total=0 chunks-cur=0 offset-head=0 offset-tail=0
[spicy-driver] memory: heap=146G fibers-cur=0 fibers-cached=0 fibers-max=0
[spicy-driver] beginning parsing input (eod=false)
[spicy] Mini::Test
[spicy-driver] parsing yielded
[spicy-driver] input : size-current=1 size-total=1 chunks-cur=1 offset-head=0 offset-tail=1
[spicy-driver] memory: heap=146G fibers-cur=1 fibers-cached=0 fibers-max=1
[spicy-driver] resuming parsing input (eod=false)
[spicy-driver] parsing yielded
[spicy-driver] input : size-current=2 size-total=2 chunks-cur=2 offset-head=0 offset-tail=2
[spicy-driver] memory: heap=146G fibers-cur=1 fibers-cached=0 fibers-max=1
[spicy-driver] resuming parsing input (eod=false)
[spicy-driver] parsing yielded
[spicy-driver] input : size-current=3 size-total=3 chunks-cur=3 offset-head=0 offset-tail=3
[spicy-driver] memory: heap=146G fibers-cur=1 fibers-cached=0 fibers-max=1
[spicy-driver] resuming parsing input (eod=false)
[spicy]   f1 = 10000
[spicy-driver] parsing yielded
[spicy-driver] input : size-current=4 size-total=4 chunks-cur=4 offset-head=0 offset-tail=4
[spicy-driver] memory: heap=146G fibers-cur=1 fibers-cached=0 fibers-max=1
[spicy-driver] resuming parsing input (eod=true)
[spicy]   f2 = 42
[spicy-driver] finished parsing input (eod=true)
[spicy-driver] input : size-current=5 size-total=5 chunks-cur=5 offset-head=0 offset-tail=5
[spicy-driver] memory: heap=147G fibers-cur=1 fibers-cached=1 fibers-max=1
```

Reply:
Yeah, we have a problem here.  It's not breaking the incremental parsing, but it's no longer trimming off data that we have fully parsed.

The change removed the low-level `trim`() calls right after the literals. I thought we'd trim later anyways, but that's not happening (just looked at the generated code). So we need to find a place to trim after a literal has been parsed, once we know it's been successful.

Reply:
Turns out that we seem to need to trim in two places since we need to support parsing of e.g., integers like in this test both when they appear naked (e.g., `x: uint8`), and also in lookahead parsing.